crates/meroctl/src/cli/config.rs

```diff
@@ -58,44 +56,46 @@ pub enum BootstrapNetwork {
 
 impl ConfigCommand {
     pub fn run(self, root_args: cli::RootArgs) -> eyre::Result<()> {
-        let path = root_args.home.join(&root_args.node_name);
+        let path = root_args
+            .home
+            .join(&root_args.node_name)
+            .join("config.toml");
 
-        let mut config = if ConfigFile::exists(&path) {
-            ConfigFile::load(&path)?
-        } else {
-            eyre::bail!("You have to initialize the node first \nRun command node init -n <NAME>");
-        };
+        // Load the existing TOML file
+        let toml_str = fs::read_to_string(&path)
+            .map_err(|_| eyre::eyre!("Node must be initialized first."))?;
+        let mut doc = toml_str.parse::<DocumentMut>()?;
 
         if self.print {
-            println!("{}", toml::to_string_pretty(&config)?);
+            println!("{}", doc.to_string());
             return Ok(());
         }
 
-        // Update boot nodes if provided
-        if !self.boot_nodes.is_empty() {
-            config.network.bootstrap.nodes.list = self.boot_nodes;
-        } else if let Some(network) = self.boot_network {
-            config.network.bootstrap.nodes.list = match network {
-                BootstrapNetwork::CalimeroDev => BootstrapNodes::calimero_dev().list,
-                BootstrapNetwork::Ipfs => BootstrapNodes::ipfs().list,
-            };
-        }
+        let ipv4_host = self.swarm_host.iter().find_map(|ip| {
+            if let IpAddr::V4(v4) = ip {
+                Some(*v4)
+            } else {
+                None
+            }
+        });
+
+        let ipv6_host = self.swarm_host.iter().find_map(|ip| {
+            if let IpAddr::V6(v6) = ip {
+                Some(*v6)
+            } else {
+                None
+            }
+        });
 
-        // Update swarm host and/or port if provided
+        // Update swarm listen addresses
         if !self.swarm_host.is_empty() || self.swarm_port.is_some() {
-            let mut new_listen = Vec::new();
-
-            let ipv4_host = self.swarm_host.iter().find_map(|ip| match ip {
-                IpAddr::V4(v4) => Some(*v4),
-                _ => None,
-            });
-
-            let ipv6_host = self.swarm_host.iter().find_map(|ip| match ip {
-                IpAddr::V6(v6) => Some(*v6),
-                _ => None,
-            });
-            for addr in config.network.swarm.listen.iter() {
+            let swarm_table = doc["swarm"].as_table_mut().unwrap();
+            let listen_array = swarm_table["listen"].as_array_mut().unwrap();
```

Comment:
I `unwrap()` here because if the config.toml is created through the cli, the swarm table and listen array are guaranteed.
But, the user could manually delete those tables/arrays which would cause a panic.
Should I cover those cases? Open to suggestions @miraclx

Reply:
how about `unwrap_or_default`?